game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -190,7 +192,7 @@ public Territory getTerritory(final String s) {
 
   private Set<Territory> getNeighbors(final Set<Territory> frontier, final Set<Territory> searched, final int distance,
       @Nullable final Predicate<Territory> cond) {
-    if (distance == 0) {
+    if (distance == 0 || frontier.isEmpty()) {
```

Comment:
Add empty check here as once the frontier is empty, you won't ever find any more territories to search.